rfcs/0012-mtls.md

```diff
@@ -0,0 +1,106 @@
+# RFC Template
+
+- Feature Name: `mtls`
+- Start Date: 2024-11-27
+- RFC PR: [Kuadrant/architecture#110](https://github.com/Kuadrant/architecture/pull/110)
+- Issue tracking: [Kuadrant/kuadrant-operator#1049](https://github.com/Kuadrant/kuadrant-operator/issues/1049)
+
+# Summary
+[summary]: #summary
+
+MTLS support for communication between the Wasm plugin in the gateway pod, and Authorino & Limitador.
+
+# Motivation
+[motivation]: #motivation
+
+For security reasons so that all traffic is encrypted, and certificates validated.
+This aligns with a 'zero trust' architecture.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Update the Kuadrant CR to have a new configuration option `mtls` for users to configure if mtls should be enabled for either authorino or limitador.
+The reason for having separate flags for either component is to allow flexibility where the resource overhead of tls may be considered too high when enabled for a specific component e.g. for checking limits on each request in limitador.
+The new config in the Kuadrant CR would look like this:
+
+```yaml
+apiVersion: Kuadrant.io/v1alpha1
+kind: kuadrant
+spec:
+  mtls:
+    authorino: true
+    limitador: true
+```
+
```

Comment:
I think it would be good to call out that if we don't support mTLS configuration for a gateway provider, it will be reflected in the status of the kuadrant resource. Also given how important this can be for users, kuadrant should probably not become ready until the specified mTLS can be handled or the mTLS section is removed.